org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/batch/NameEnvironmentAnswerWithClasspathOrigin.java

```diff
@@ -0,0 +1,47 @@
+package org.eclipse.jdt.internal.compiler.batch;
+
+import org.eclipse.jdt.internal.compiler.batch.FileSystem.Classpath;
+import org.eclipse.jdt.internal.compiler.env.AccessRestriction;
+import org.eclipse.jdt.internal.compiler.env.IBinaryType;
+import org.eclipse.jdt.internal.compiler.env.ICompilationUnit;
+import org.eclipse.jdt.internal.compiler.env.ISourceType;
+import org.eclipse.jdt.internal.compiler.env.NameEnvironmentAnswer;
+import org.eclipse.jdt.internal.compiler.lookup.ModuleBinding;
+import org.eclipse.jdt.internal.compiler.lookup.ReferenceBinding;
+
+public class NameEnvironmentAnswerWithClasspathOrigin extends NameEnvironmentAnswer {
```

Comment:
@jukzi I don't like a separate class. I wonder if we can have another interface next to `NameEnvironmentAnswer` and just add this field there. All I need is the `Path` of the jar or directory (in terms of batch compiler). But I can image that an IDE would want to have more interesting objects here.

Reply:
> I wonder if we can have another interface next to `NameEnvironmentAnswer` (like `IClasspathOrigin`) and just add this field directly in `NameEnvironmentAnswer`

Sounds good.

Reply:
That would also remove unneeded change in jdt.core